assets/js/modules/adsense/components/module/ModuleOverviewWidget/index.js

```diff
@@ -0,0 +1,142 @@
+
+	const currentRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate,
+		endDate,
+	};
+	const previousRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate: compareStartDate,
+		endDate: compareEndDate,
+	};
+	const currentRangeChartArgs = {
+		...currentRangeArgs,
+		dimensions: [ 'DATE' ],
+	};
+	const previousRangeChartArgs = {
+		...previousRangeArgs,
+		dimensions: [ 'DATE' ],
+	};
+
+	const currentRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( currentRangeArgs ) );
+	const previousRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( previousRangeArgs ) );
+	const currentRangeChartData = useSelect( ( select ) => select( STORE_NAME ).getReport( currentRangeChartArgs ) );
+	const previousRangeChartData = useSelect( ( select ) => select( STORE_NAME ).getReport( previousRangeChartArgs ) );
+
+	const currentRangeLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ currentRangeArgs ] ) );
+	const previousRangeLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ previousRangeArgs ] ) );
+	const currentRangeChartLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ currentRangeChartArgs ] ) );
+	const previousRangeChartLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ previousRangeChartArgs ] ) );
+
+	const currentRangeError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ currentRangeArgs ] ) );
+	const previousRangeError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ previousRangeArgs ] ) );
+	const currentRangeChartError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ currentRangeChartArgs ] ) );
+	const previousRangeChartError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ previousRangeChartArgs ] ) );
+
+	const handleStatsSelection = useCallback( ( stat ) => {
+		setSelectedStats( stat );
+	}, [] );
+
+	if ( currentRangeLoading || previousRangeLoading || currentRangeChartLoading || previousRangeChartLoading ) {
+		return <ProgressBar />;
+	}
+
+	if ( currentRangeError || previousRangeError || currentRangeChartError || previousRangeChartError ) {
+		return (
+			<WidgetReportError
+				moduleSlug="adsense"
+				error={ currentRangeError || previousRangeError || currentRangeChartError || previousRangeChartError }
+			/>
+		);
+	}
+
+	if ( isZeroReport( currentRangeData ) || isZeroReport( currentRangeChartData ) ) {
+		return <WidgetReportZero moduleSlug="adsense" />;
+	}
+
+	return (
+		<Widget
+			noPadding
+			Header={ Header }
+		>
+			<Overview
+				metrics={ metrics }
+				currentRangeData={ currentRangeData }
+				previousRangeData={ previousRangeData }
+				selectedStats={ selectedStats }
+				handleStatsSelection={ handleStatsSelection }
```

Comment:
We can pass the setter here instead which is stable from `useState` – no need for an added `useCallback` around it 👍 

```suggestion
				handleStatsSelection={ setSelectedStats }
```

Reply:
Nice. Updated!